telegram/ext/filters.py

```diff
@@ -120,17 +131,49 @@ class MergedFilter(BaseFilter):
         or_filter: Optional filter to "or" with base_filter. Mutually exclusive with and_filter.
 
     """
+    update_filter = True
 
     def __init__(self, base_filter, and_filter=None, or_filter=None):
         self.base_filter = base_filter
+        if self.base_filter.regex_filter:
+            self.regex_filter = True
         self.and_filter = and_filter
+        if self.and_filter and not isinstance(self.and_filter,
+                                              bool) and self.and_filter.regex_filter:
+            self.regex_filter = True
         self.or_filter = or_filter
+        if self.or_filter and not isinstance(self.and_filter,
+                                             bool) and self.or_filter.regex_filter:
+            self.regex_filter = True
 
-    def filter(self, message):
+    def filter(self, update):
+        base_output = self.base_filter(update)
         if self.and_filter:
-            return self.base_filter(message) and self.and_filter(message)
+            comp_output = self.and_filter(update)
+            if base_output and comp_output:
```

Comment:
I would love a couple of comments on this code... Was a bit confusing at first to me :)

Reply:
Added a comment on top and simplified code